### PR TITLE
refactor(api): application variables crypto 

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12642.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

File: engine/api/application/dao_variable.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	query := `DELETE FROM application_variable
+		  WHERE application_variable.application_id = $1 AND application_variable.var_name = $2`
+	result, err := db.Exec(query, appID, variable.Name)
+	if err != nil {
+		return sdk.WrapError(err, "Cannot delete variable %s", variable.Name)
+	}
+
+	rowAffected, err := result.RowsAffected()
+	if err != nil {
+		return err
```
Comment (richardlt) [inline engine/api/application/dao_variable.go]: Wrap

File: engine/api/application/dao_variable.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	}
+	if rowAffected == 0 {
+		return ErrNoVariable
```
Comment (richardlt) [inline engine/api/application/dao_variable.go]: Wrap

File: engine/api/application/dao_variable.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+// InsertVariable Insert a new variable in the given application
+func InsertVariable(db gorp.SqlExecutor, appID int64, v *sdk.Variable, u sdk.Identifiable) error {
+	//Check variable name
+	rx := sdk.NamePatternRegex
+	if !rx.MatchString(v.Name) {
+		return sdk.NewError(sdk.ErrInvalidName, fmt.Errorf("Invalid variable name. It should match %s", sdk.NamePattern))
+	}
+
+	if sdk.NeedPlaceholder(v.Type) && v.Value == sdk.PasswordPlaceholder {
+		return fmt.Errorf("You try to insert a placeholder for new variable %s", v.Name)
```
Comment (richardlt) [inline engine/api/application/dao_variable.go]: Wrap

File: engine/api/application/gorp_model.go [hunk 4]
```diff
@@ -134,14 +187,12 @@ func (ava *dbApplicationVariableAudit) PostUpdate(db gorp.SqlExecutor) error {
...
-	if ava.VariableAfter != nil {
-		v, err := json.Marshal(ava.VariableAfter)
-		if err != nil {
-			return err
-		}
-		vA.Valid = true
-		vA.String = string(v)
+	v, err := json.Marshal(ava.VariableAfter)
+	if err != nil {
+		return err
```
Comment (richardlt) [inline engine/api/application/gorp_model.go]: Wrap

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12701.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**